truffle/src/main/java/org/jruby/truffle/nodes/methods/ExceptionTranslatingNode.java

```diff
@@ -134,7 +134,11 @@ private RubyException translate(UnsupportedSpecializationException exception) {
                     }
                 }
             } else {
-                builder.append(value.getClass().getName());
+                if (value == null) {
+                    builder.append("null");
```

Comment:
I met the same problem recently, but I did the check at line 103, which might to follow the flow better.